11-bug_tracking.Rmd

```diff
@@ -1,23 +1,127 @@
-# Bug Tracking
-
-## Using the Bug Tracker (Bugzilla)
-
-### Checking if a bug already exists
-
-### Reporting a bug
-
-### Understanding the issue’s progress and status
-
-## Disagreement With a Resolution on the Bug Tracker
-
-## Helping Triage Issues
-
-### Classifying Reports
-
-### Reviewing Patches
-
-### Finding an Issue You Can Help With
-
-## Gaining the “Developer” Role on the Bug Tracker
-
-## The Meta Tracker
+# Bug Tracking
+
+## What is a bug in $\textsf{R}$ ?
+
+You may find a bug in $\textsf{R}$ if:
+
+
+1. The $\textsf{R}$ session terminates unexpectedly, or there is a segmentation fault, it might be a bug, unless you have written your own call to compiled code or an internal function (via `.C` or `.Internal`). The error may look like this:
+
+```{r, results='hide'}
+##  *** caught segfault ***
+## address (nil), cause 'memory not mapped'
+```
+
+2. If the code does not do what the documentation says it should, then either the code or the documentation is wrong. Report either of which needs to be fixed.
+
+$\textbf{Note}$:
+When you are in doubt that there is a bug:
+
+1. Make sure whether the bug appears in a clean session of $\textsf{R}$. Many a times, there are variables/commands/functions stored in history which might causes issues. Hence, check in a clean session. To do so, launch $\textsf{R}$ from the command line with the `--vanilla` option. 
+
+2. At times the code that is written is very complicated, has numerous package and file dependencies, has many function calls, etc.. In such scenarios it is quite common that the code throws an error and you are not able to solve it. You may tend to think that there is a bug that needs to be reported. Before doing so, try to produce a minimum working example of the code for the section where the error occurred. Add only those package and files which are required by that section, and see if the error appears still. Using this approach shall solve most of the errors.
+
+## What may not be a bug?
+
+1. In case the code is doing something unexpected, it may not necessarily be a bug. Carefully review the documentation of the function being called, and check whether the behaviour being exhibited on calling this function is the same as it was designed to do.
+
+2. Issues with $\textit{seemingly}$ identical numbers not being equal (especially floating point numbers) are usually not bugs. 
+
+3. If $\textsf{R}$ is running slower than expected, then also it may not be a bug. Ask someone else to review your code in such a case.
+
+4. If some function is working, but it is not defined in the best generalised way, then consult some to look over your code. This may perhaps not be a bug, instead it might be an alternative way of writing the function.

```

Comment:
```suggestion
4. If some function is working, but it is not defined in the best generalised way, then consult someone to look over your code. This may perhaps not be a bug; instead, it might be an alternative way of writing the function.

```